configs/local_setup.yml

```diff
@@ -26,5 +26,7 @@
   "log-dir": "logs",
   "use_wandb": True,
   "wandb_host": "https://api.wandb.ai",
-  "wandb_project": "neox"
+  "wandb_project": "neox",
+  "num_gpus": 1,
+  "ia3_prompt_tuning": True
```

Comment:
I've been thinking a nice pattern for the GPT-NeoX configurations is to make them composable so you can setup different combinations using the script syntax in this [experiment script](https://github.com/OpenBioML/chemnlp/tree/main/experiments#working-with-the-stability-cluster) without having the touch the core configs of cluster / model;
* cluster config
* model config
* prompt tuning config (additional option)
* (IA)3 config (additional option)

The way soft prompt tuning is handled in GPT-NeoX has a nested dictionary of further PEFT-related options. Are there any equivalent options for around initialisation, etc (IA)3?
```
{
  # prompt tuning settings
   "soft_prompt_tuning": {
      "enabled": True, # also freezes all other parameters
      "n_tokens": 10,
      "init_string": "",
      "init_range": 0.5,
   }
}
```

Also, in theory I guess there's nothing stopping us from using both (IA)3 + prompt tuning at the same time.

Reply:
In the paper the just initialize the IA3 vectors with 1s (https://arxiv.org/pdf/2205.05638.pdf#page=6 passage below the formula in the middle):
> lk, lv, and lff areallinitializedwithonesso that the overall function computed by the model does not change when they are added.

I'm not sure if adding some noise to the vectors could make sense? But likely not.

Reply:
Maybe rename `ia3_prompt_tuning` to `ia3_tuning` because the call the vectors adapters and promp tuning is related but different?
But yeah, this is just a minor detail. 😃

Reply:
> The way soft prompt tuning is handled in GPT-NeoX has a nested dictionary of further PEFT-related options. Just wondering, are there any equivalent options for around initialisation, etc for (IA)3?
> 
> ```
> {
>   # prompt tuning settings
>    "soft_prompt_tuning": {
>       "enabled": True, # also freezes all other parameters
>       "n_tokens": 10,
>       "init_string": "",
>       "init_range": 0.5,
>    }
> }
> ```

There are no hyperparameters given we're not adding the extra loss terms.